### DISCUSSION TITLE
message when worker exits

### DISCUSSION THREAD

**Discussion by kollol-chowdhury-incontact** [Kong/kong#8782] [Help]
Like we have init method for worker,do we have something like exit method to cleanup all activities before worker process exits?

TIA

**Comment by jschmid1** [comment]
There was a discussion about a `exit_worker` method that could do that job. In order to justify this - what would you expect from this function?

**Comment by kollol-chowdhury-incontact** [comment]
Hi @jschmid1 , I got the solution to this.
I am collecting logs and sending it based on an interval.
So right before it exits,if there is any data,it should send it..

**Reply by hanshuebner** [reply]
Hello @kollol-chowdhury-incontact,

thank you for the description of your use case.  At this point, Kong does not have the hook that you're looking for.  Generally speaking, having shutdown hooks always bears the danger that the shutdown actions are not executed when the system shuts down ungracefully.   Maybe you can work around the requirement in a different way?

Kind regards,
Hans

**Comment by Tieske** [comment]
we actually have this exact problem, see https://github.com/Kong/kong/blob/master/kong/tools/batch_queue.lua#L143-L146 

If the timer executes prematurely we exit, instead of flushing whatever we have in the queue. This causes data loss (but only logs).

**Comment by hanshuebner** [comment]
A nice hack that was suggested by @Tieske is to create a [timer](https://kong.github.io/lua-resty-timer/index.html) to fire in the very far future, but defined a `cancel` callback.  The callback will be invoked when Kong is orderly shut down, so it can be used to perform specific cleanups.  This might be a viable workaround to fulfil your requirement.

Kind regards,
Hans
